### PR TITLE
Implement MUI UI Library, adjust theme, and add checkbox

### LINKED ISSUE #31: 15. As a user, I want the appearance of the app to be consistent and cohesive.
## Summary
Using a quality UI framework can speed production and bring cohesion to your web app's aesthetic. Our group expressed interest in using Material UI. Your task is to add the framework as a dependency and ensure that it is working by importing the MUI checkmark and using it in the ListItem component.

## Acceptance criteria
- [ ] [Install MUI and related dependencies](https://mui.com/material-ui/getting-started/installation/#npm)
`npm install @mui/material @emotion/react @emotion/styled
`
- [ ] Install Roboto (or font of your choosing) using Google Fonts CDN, by adding the following to the project `<Head>` tag: 
`<link
  rel="stylesheet"
  href="https://fonts.googleapis.com/css?family=Roboto:300,400,500,700&display=swap"
/>` and ensure that it is working properly.

- [ ] Install Material Icon font
`npm install @mui/icons-material` and ensure that you can successfully add an icon. Note: we will eventually want the search icon and a delete icon.
- [ ] Replace checkmark in ListItem with MUI checkmark component

### A Stretch Goal
Implement a basic theme. The checkmark is only used in this file, however, buttons and other UI elements that are used in multiple places, ideally have style overrides set in one place rather than in each instance.
- [ ] Make `theme.js` file in `src/themes`
- [ ] Export basic theme changing primary color to the primary green used throughout the project. 
- [This theme playground may be helpful](https://bareynol.github.io/mui-theme-creator/)
- [MUI theme docs](https://mui.com/material-ui/customization/theming/)

## Notes
- For the [checkmark component](https://mui.com/material-ui/react-checkbox/), you will want to reference the documentation for 'controlled' checkmarks as we are using `checked` and `onChange`

### PR DESCRIPTION
**Pull request by wheninseattle**
## Description

This issue introduces the MUI UI framework to the project. It also establishes a theme and implements a few MUI components including the Checkbox and a search bar that utilizes icon adornment. 

## Related Issue

closes #31 

## Acceptance Criteria

- [x] [Install MUI and related dependencies](https://mui.com/material-ui/getting-started/installation/#npm)
`npm install @mui/material @emotion/react @emotion/styled
`
- [x] Install Roboto (or font of your choosing) using Google Fonts CDN, by adding the following to the project `<Head>` tag: 
`<link
  rel="stylesheet"
  href="https://fonts.googleapis.com/css?family=Roboto:300,400,500,700&display=swap"
/>` and ensure that it is working properly.

- [x] Install Material Icon font
`npm install @mui/icons-material` and ensure that you can successfully add an icon. Note: we will eventually want the search icon and a delete icon.
- [x] Replace checkmark in ListItem with MUI checkmark component

### A Stretch Goal
Implement a basic theme. The checkmark is only used in this file, however, buttons and other UI elements that are used in multiple places, ideally have style overrides set in one place rather than in each instance.
- [x] Make `theme.js` file in `src/themes`
- [x] Export basic theme changing primary color to the primary green used throughout the project. 

## Type of Changes

|     | Type                       |
| --- | -------------------------- |
|     | :bug: Bug fix              |
| ✓   | :sparkles: New feature     |
|     | :hammer: Refactoring       |
|     | :100: Add tests            |
| ✓    | :link: Update dependencies |
|     | :scroll: Docs              |

### Before

Before

### After

After

## Testing Steps / QA Criteria

1. Git Pull WN-MUI-Integration
2. run `npm i` in terminal to install new dependencies 
3. run `npm start` to ensure that the UI changes reflect the 'After' image shown above
4. Start having fun with MUI!

### CODE REVIEW

**Review by tmflower — APPROVED** [review]
Looking forward to my crash course in MUI!

**Review by mariahschock — APPROVED** [review]
Nice work! Already looks cleaner. Excited to see what else we can do with it!

**Review by a-cole55 — APPROVED** [review]
Great job Weston! One question/comment... what are your thoughts on the name being "HoneyDo" instead of Honeydew? I think originally that's what you proposed but I think it's a cool play on words since this is technically a "to do"/to buy shopping list.

File: src/components/SearchBar.jsx [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+					<InputAdornment position="start">
+						<Search />
+					</InputAdornment>
+				}
+				endAdornment={
+					searchTerm.length ? (
+						<InputAdornment position="end">
+							<HighlightOff onClick={onSearchReset} type="reset" />
+						</InputAdornment>
+					) : null
```
Comment (YolandaHaynes) [inline src/components/SearchBar.jsx:36]: I haven't used Material UI, but I really like this logic! I just notice that when you hoover over the HighlightOff icon, the cursor doesn't switch to an arrow, so a user might be confused if they can even click on that icon. If you have the time and bandwidth, see if you can add that arrow cursor. Other than that great job!

File: index.html [hunk 2]
```diff
@@ -12,7 +12,11 @@
 		/>
 		<link rel="icon" type="image/svg+xml" href="/src/favicon.ico" />
 		<meta name="theme-color" content="#000000" />
-		<title>Smart Shopping List</title>
+		<title>Honeydew | Smart Shopping List</title>
```
Comment (YolandaHaynes) [inline index.html:15]: I like the new app name! The team's product is coming along nicely!!!

File: src/views/List.jsx [hunk 3]
```diff
@@ -38,21 +27,7 @@ export function List({ data, listToken }) {
...
-							placeholder="Start typing here..."
-							onChange={onSearch}
-						/>
-						{!!searchTerm && (
-							<button onClick={onSearchReset} type="reset">
-								Reset
-							</button>
-						)}
-					</form>
+					<SearchBar searchTerm={searchTerm} setSearchTerm={setSearchTerm} />
```
Comment (YolandaHaynes) [inline src/views/List.jsx:30]: Love that you created a new component! It's cleaner, and it makes good use of compartmentalizing the search feature.

**Review by YolandaHaynes — APPROVED** [review]

**Pull request merged**